src/index/TextIndexLiteralFilter.h

```diff
@@ -0,0 +1,69 @@
+// Copyright 2025, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Felix Meisen (fesemeisen@outlook.de)
+
+#ifndef QLEVER_SRC_INDEX_TRIPLEINTEXTINDEX_H
+#define QLEVER_SRC_INDEX_TRIPLEINTEXTINDEX_H
+
+#include <string>
+
+#include "parser/RdfParser.h"
+
+class TextIndexLiteralFilter {
+ public:
+  enum struct FilterType { AcceptMatching, DeclineMatching };
+
+  TextIndexLiteralFilter()
+      : regex_{std::make_unique<RE2>("(?s).*")},
+        filterType_{FilterType::AcceptMatching} {};
+
+  /**
+   * @brief Class to determine whether the literal of a triple should be part of
+   *        the text index or not.
+   * @param regex The regex that is used to determine whether a literal should
+   *              be part of the text index or not. This is used on the
+   *              predicate.
+   * @param filterType If set to AcceptMatching, the matching regex cases will
```

Comment:
```suggestion
   * @param filterType If set to `AcceptMatching`, the matching regex cases will
```
(similar for other mentions of types and variable names in comments).